assets/js/custom/Translation.js

```diff
@@ -2,6 +2,111 @@
 
 // eslint-disable-next-line no-unused-vars
 function Translation (hasDescription, hasCredit) {
+  let languageMap = {}
+  let targetLanguage = document.documentElement.lang
+
+  $(document).ready(function () {
+    $.ajax({
+      url: '../languages',
+      type: 'get',
+      success: function (data) {
+        languageMap = data
+      }
+    })
+  })
+
+  $('.mdc-list-item__text').each(function () {
+    if ($(this).parent()[0].hasAttribute('selected')) {
+      targetLanguage = fixLanguageCode($(this).parent().attr('data-value'))
+    }
+  })
+
+  $(document).on('click', '.comment-translation-button', function () {
+    const commentId = $(this).attr('id').substring('comment-translation-button-'.length)
+
+    $(this).hide()
+
+    if ($('#comment-text-translation-' + commentId).attr('lang') !== targetLanguage) {
+      $('#comment-translation-loading-spinner-' + commentId).show()
+      translateComment(commentId)
+    } else {
+      openTranslatedComment(commentId)
+    }
+  })
+
+  $(document).on('click', '.remove-comment-translation-button', function () {
+    const commentId = $(this).attr('id').substring('remove-comment-translation-button-'.length)
+    $(this).hide()
+    $('#comment-translation-button-' + commentId).show()
+    $('#comment-translation-wrapper-' + commentId).slideUp()
+    $('#comment-text-wrapper-' + commentId).slideDown()
+  })
+
+  function fixLanguageCode (languageCode) {
+    if (languageCode.length > 2) {
+      languageCode = languageCode.split('')
+      languageCode[2] = '-'
+      languageCode = languageCode.join('')
+    }
+
+    return languageCode
+  }
+
+  function getTranslationProvider (provider) {

```

Comment:
This function can be replaced with a map. I think it's ok to not consider the case where the provider is not recognized

Reply:
Replaced with a map as suggested. Good idea